lib/std/crypto/kyber_d00.zig

```diff
@@ -1020,8 +1020,14 @@ const Poly = struct {
                 //                  = ⌊(2ᵈ/q)x+½⌋ mod⁺ 2ᵈ
                 //                  = ⌊((x << d) + q/2) / q⌋ mod⁺ 2ᵈ
                 //                  = DIV((x << d) + q/2, q) & ((1<<d) - 1)
-                const t = @as(u32, @intCast(p.cs[in_off + i])) << d;
-                in[i] = @as(u16, @intCast(@divFloor(t + q_over_2, Q) & two_d_min_1));
+                const t = @as(u24, @intCast(p.cs[in_off + i])) << d;
+                // Division by invariant multiplication, equivalent to DIV(t + q/2, q).
+                // A division may not be a constant-time operation, even with a constant denominator.
+                // Here, side channels would leak information about the shared secret, see https://kyberslash.cr.yp.to
+                // Multiplication, on the other hand, is a constant-time operation on the CPUs we currently support.
+                comptime assert(((20642679 * @as(u64, Q)) >> 36) == 1);
+                const u: u32 = @intCast((@as(u64, t + q_over_2) * 20642679) >> 36);
```

Comment:
Just checked this is correct for `0 ≤ t < 41520952`, viz `d ≤ 13`. Maybe we should add a static assert that `d ≤ 11` anyway.

Reply:
> Just checked this is correct for `0 ≤ t < 41520952`, viz `d ≤ 13`. Maybe we should add a static assert that `d ≤ 11` anyway.

Sure, static asserts are free and never hurt!